app/V1Module/presenters/GroupsPresenter.php

```diff
@@ -852,6 +875,47 @@ public function actionStudentsStats(string $id, string $userId)
         $this->sendSuccessResponse($this->groupViewFactory->getStudentsStats($group, $user));
     }
 
+    public function checkStudentsSolutions(string $id, string $userId)
+    {
+        $user = $this->users->findOrThrow($userId);
+        $group = $this->groups->findOrThrow($id);
+
+        // actually studentStats permission is like a soft pre-condition, the solutions are filtered individually
+        if (!$this->groupAcl->canViewAssignments($group) || !$this->groupAcl->canViewStudentStats($group, $user)) {
+            throw new ForbiddenRequestException();
+        }
+    }
+
+    /**
+     * Get all solutions of a single student from all assignments in a group
+     * @GET
+     * @param string $id Identifier of the group
+     * @param string $userId Identifier of the student
+     * @throws BadRequestException
+     */
+    public function actionStudentsSolutions(string $id, string $userId)
+    {
+        $user = $this->users->findOrThrow($userId);
+        $group = $this->groups->findOrThrow($id);
+
+        if ($group->isStudentOf($user) === false) {
```

Comment:
Isn't this checked in `check` method using ACL?

Reply:
Thinking about this... even if it is, having this check and returning 400 makes sense...